lib/web/apiserver.go

```diff
@@ -2648,6 +2648,37 @@ func makeUnifiedResourceRequest(r *http.Request) (*proto.ListUnifiedResourcesReq
 	}, nil
 }
 
+// checkerWithLogins is a [services.AccessChecker] that augments the allowed
+// logins for a resource based on information received from leaf clusters.
+type checkerWithLogins struct {
+	leafLogins []string
+	services.AccessChecker
+}
+
+// GetAllowedLoginsForResource overrides the wrapped [services.AccessChecker] to return logins
+// that are allowed in both the leaf and root cluster.
+func (c checkerWithLogins) GetAllowedLoginsForResource(resource services.AccessCheckable) ([]string, error) {
+	localLogins, err := c.AccessChecker.GetAllowedLoginsForResource(resource)
```

Comment:
The wrapped checker is a bit weird and I'm not sure it's entirely valid to all GetAllowedLoginsForResource in the root for a leaf resource. Can we get rid of the checker, use WithLogins in both root and leaf, and filter by identity.Principals instead of the GetAllowedLoginsForResource? I guess we might need to fall back to the checker if the leaf doesn't support WithLogins

Reply:
yeah it looks like ui.MakeServer at least only needs the accesschecker to get the logins - i guess for root clusters maybe it's not obvious whether it's better to keep calling `GetAllowedLoginsForResource` in the proxy or shift it to auth, but probably more consistent to just shift it to auth for both root and leaf cluster. I definitely don't think we should be calling it on the root proxy for leaf resources, I thought that's what we were avoiding with this pr, should be able to use the tls identity.Principals instead